ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -751,7 +751,7 @@ static char *make_filename_safe(const char *filename)
 		}
 		return estrdup(filename);
 	}
-	if (*filename && memcmp(filename, ":memory:", sizeof(":memory:"))) {
+	if (*filename && zend_binary_strcmp(filename, strlen(filename)-1, ":memory:", strlen(":memory:")-1)) {
```

Comment:
I know very little pdo sqlite, but I do not get why -1 for the 2 strlen ? Legit question :)